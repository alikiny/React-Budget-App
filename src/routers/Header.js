import React from 'react';
import { NavLink } from 'react-router-dom';
import {connect} from 'react-redux'
import {logoutAction} from '../firebase/logOutAction'

export const Header = ({logoutAction}) => (
    <div className='header'>
        <h1 className="app-name">Budget Management</h1>
        <div className="navbar navbar-text navbar-expand-lg">
            <ul className="navbar-nav">
                <li className="nav-item">
                    <NavLink to='dashboard' className='nav-link' activeClassName="link-active">Dashboard</NavLink>
                </li>

                <li className="nav-item">
                    <NavLink to='expenses' className='nav-link' activeClassName="link-active">Expenses</NavLink>
                </li>

                <li className="nav-item">
                    <NavLink to='summary' className='nav-link' activeClassName="link-active">Summary</NavLink>
                </li>

                <li className="nav-item">
                    <button type="button" onClick={logoutAction}>Log out</button>
                </li>
            </ul>
        </div>
    </div>
)


const mapDispatchToProp=(dispatch)=>({
    logoutAction:()=>dispatch(logoutAction())
})
export default connect(undefined,mapDispatchToProp)(Header)