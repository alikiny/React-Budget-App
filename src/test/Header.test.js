import ShallowRenderer from 'react-test-renderer/shallow'
import React from 'react'
import Header from '../routers/Header'
import {shallow} from 'enzyme'

// test('Header render correctly',()=>{
//     const renderer=new ShallowRenderer()
//     renderer.render(<Header/>)
//     expect(renderer.getRenderOutput()).toMatchSnapshot()

// })

test('Header render correctly',()=>{
    const wrapper= shallow(<Header/>)
    expect(wrapper).toMatchSnapshot()
})

