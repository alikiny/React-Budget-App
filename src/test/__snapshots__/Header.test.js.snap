// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<div
  className="header"
>
  <h1>
    <NavLink
      activeClassName="link-active"
      className="nav-link app-name"
      exact={true}
      to="/"
    >
      Budget Manager
    </NavLink>
  </h1>
  <div
    className="navbar navbar-text navbar-expand-lg"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="link-active"
          className="nav-link"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="link-active"
          className="nav-link"
          to="dashboard"
        >
          Dashboard
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="link-active"
          className="nav-link"
          to="expenses"
        >
          Expenses
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="link-active"
          className="nav-link"
          to="summary"
        >
          Summary
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <button
          type="button"
        >
          Log out
        </button>
      </li>
    </ul>
  </div>
</div>
`;
